meta-mender-core/classes/mender-helpers.bbclass

```diff
@@ -19,6 +22,9 @@ get_uboot_device_from_device() {
         /dev/mmcblk[0-9])
             echo ${1#/dev/mmcblk}
             ;;
+        ubi[0-9])
+            echo ubi
```

Comment:
> I suppose since a specification without /dev is so common, it makes sense to have MENDER_STORAGE_DEVICE refer to ubi0 directly.

Agreed, and this was reason that I initially suggested that.

> Rename MENDER_UBOOT_STORAGE_DEVICE to MENDER_UBOOT_STORAGE_DEVICE_NUMBER, not just for ubi, but for existing sdimg as well. A number is what it has been all along, so this is more accurate and descriptive.

I can do this to make the it accurate and descriptive, but will not help my case. Explanation below.

> Similarly, rename get_uboot_device_from_device to get_uboot_device_number_from_device and have it return ${1#ubi}, IOW just a number.

This is where I would run in to problems. There is no way to parse this number (MTD) from the "volume" table (ubi0, ubi0_X), which this would represent. That is why I want to create the "`MENDER_MTD_UBI_DEVICE`"

> Taken together, the previous two changes will make mender_uboot_dev into a number. Now replace:

> ubi part ${mender_uboot_dev}
> with:

> ubi part ${mender_uboot_if}${mender_uboot_dev}

Sadly the `ubi part` command only takes a name as argument (ubi) in my case.

> For Mender opening the ubi block device, we got two choices:

> Add the reference /dev/ when rendering the config file in mender_0.1.bb.

> Have the Mender client itself accept bare references to ubi*, and transparently add the right path in the code. I can help with this, if you need it.

> I think I vote for the second option, since the Mender client appears to be the only component in this puzzle that isn't capable of handling bare ubi references.

I vote second option as well, and was my initial idea.

> Create the U-Boot variable mender_uboot_root_name as an alias to mender_kernel_root_name. This has no functional impact, it's just to have unambiguous variables for uboot and kernel references, respectively.

Yes. I like this, was my though as well.

> Replace:

> ubifsmount ${mender_kernel_root_name}
> with:

> ubifsmount ${mender_uboot_root_name}

Yes on this one.

I will try and summarize my TODO list.

- stick to `ubi0` as MENDER_STORAGE_DEVICE

- WONT rename `MENDER_UBOOT_STORAGE_DEVICE` since this does not really help me out, we could do this in a separate pull-request?

- Create the U-Boot variable `mender_uboot_root_name` as an alias to `mender_kernel_root_name`

- Create `MENDER_MTD_UBI_DEVICE_NAME` and `mender_mtd_ubi_dev_name`, since `ubi part` only takes a name.

- Remove UBI support from mender-helpers? I wont be using ${mender_uboot_boot}, override to `dummy` in mender-install-ubi to prevent it to try to parse it?

And I would end up with the following usage in u-boot:

`ubiargs=ubi.mtd=${mender_mtd_ubi_dev_name} root=${mender_kernel_root} rootfstype=ubifs`

`ubi part ${mender_mtd_ubi_dev_name}`

`ubifsmount ${mender_uboot_root_name}`

If you agree with above I will do a re-spin and we can continue the discussions.